### PR TITLE
Implement Get Element Attribute.

### LINKED ISSUE #25: feat: Add Get Element Attribute implimatnion

### PR DESCRIPTION
**Pull request by grokys**
Implements `/session/{session id}/element/{element id}/attribute/{name}`.

Not so crazy about using reflection here, but it seems it's the only way to get the property ID from a string?

Fixes #25 

## TODO

- [x] Add test
- [x] Make test pass
- [x] Add support for `Pattern.Property` attributes as supported by WinAppDriver (e.g. https://github.com/microsoft/WinAppDriver/blob/master/Tests/WebDriverAPI/ElementAttribute.cs#L69)

### CODE REVIEW

**Comment by grokys** [comment]
Yep, confirmed that the problem is the `Selenium.WebDriver` package. Changing to use `<PackageReference Include="Appium.WebDriver" Version="5.0.0-rc.8" />` allows the tests to run.

**Comment by aristotelos** [comment]
> Yep, confirmed that the problem is the `Selenium.WebDriver` package. Changing to use `<PackageReference Include="Appium.WebDriver" Version="5.0.0-rc.8" />` allows the tests to run.

I did a quick test, `GetAttribute` indeed converts to a script. But you can use `GetDomAttribute` instead which maps to the attribute API. 

I am OK converting the tests to use Appium.WebDriver, but I think keeping the freedom to use Selenium.WebDriver as a user is also good. Can we document both options in the README? I.e. we should document that Selenium.WebDriver's `GetAttribute` is not supported but `GetDomAttribute` is. (I am also still wondering if there is a Selenium option to use the attribute API when calling `GetAttribute`.)

**Comment by aristotelos** [comment]
Can we add a little explanation to the README about attributes and properties and their difference (if there is any)? Adding it to the [WebDriver interpretation](https://github.com/FlaUI/FlaUI.WebDriver?tab=readme-ov-file#webdriver-interpretation) section would make sense to me. @Roemer Do you have suggestions of attribute vs. property interpretation, is there a difference or are they the same for Windows UI automation elements?

File: src/FlaUI.WebDriver.UITests/ElementTests.cs [hunk 1]
```diff
@@ -210,5 +210,18 @@ public void ActiveElement_Default_IsSupported()
 
             Assert.That(activeElement.Text, Is.EqualTo("Invoked!"));
         }
+
+        [TestCase(["ClassName", "TextBox"])]
```
Comment (aristotelos) [inline src/FlaUI.WebDriver.UITests/ElementTests.cs:214]: Shouldn't we add a test case for a non-existing attribute that it returns `null`?

**Comment by grokys** [comment]
> I did a quick test, GetAttribute indeed converts to a script. But you can use GetDomAttribute instead which maps to the attribute API.

Ah! Yes that works! Ok then I think #28 is unnecessary for the moment then. I've pushed a change to the test to use `GetDomAttribute`.

File: src/FlaUI.WebDriver.UITests/ElementTests.cs (hunk 1 above)
Comment (grokys) [reply src/FlaUI.WebDriver.UITests/ElementTests.cs:214]: Done.

**Comment by grokys** [comment]
Added support for pattern properties. I think this PR should be ready for review now.

> Can we add a little explanation to the README about attributes and properties and their difference (if there is any)? Adding it to the [WebDriver interpretation](https://github.com/FlaUI/FlaUI.WebDriver?tab=readme-ov-file#webdriver-interpretation) section would make sense to me

Sure, though I'm not sure I'm the right person to write that ;) From what I understand in WinAppDriver, attributes simply map to UIA element properties and pattern properties if they're in the form `Pattern.Property`.

> Is there a documentation from Microsoft on which attributes are supported on Windows UI automation elements? If so, it would be good to refer to it in the README.

I'm not sure there is... I can't remember where I found out about it but I assume it was StackOverflow.

**Comment by aristotelos** [comment]
@Roemer Is this the right link to reference for UIA3 automation element properties? https://learn.microsoft.com/en-us/windows/win32/winauto/uiauto-automation-element-propids

**Comment by grokys** [comment]
@aristotelos you didn't ask me, but I believe that is the correct link for the UAI3 properties, yes.

**Comment by aristotelos** [comment]
@grokys Created https://github.com/grokys/FlaUI.WebDriver/pull/1 for updating the documentation a little. 

Could you also look into the build warnings generated about using nullable reference types?

**Review by aristotelos — CHANGES_REQUESTED** [review]
Fix the build check warnings: "The annotation for nullable reference types should only be used in code within a '#nullable' annotations context."

**Comment by grokys** [comment]
> Fix the build check warnings: "The annotation for nullable reference types should only be used in code within a '#nullable' annotations context."

Fixed. Any reason the tests aren't using `nullable`?

**Comment by aristotelos** [comment]
> > Fix the build check warnings: "The annotation for nullable reference types should only be used in code within a '#nullable' annotations context."
> 
> Fixed. Any reason the tests aren't using `nullable`?

No, no good reason, so a PR to change that is welcome.

**Comment by aristotelos** [comment]
@grokys Could you still merge https://github.com/grokys/FlaUI.WebDriver/pull/1 into your branch so that the documentation is more complete?

**Comment by grokys** [comment]
@aristotelos grokys#1 merged, thanks for doing that!

File: src/FlaUI.WebDriver.UITests/ElementTests.cs [hunk 2]
```diff
@@ -240,5 +240,39 @@ public void ActiveElement_Default_IsSupported()
...
+            using var driver = new RemoteWebDriver(WebDriverFixture.WebDriverUrl, driverOptions);
+            var element = driver.FindElement(ExtendedBy.AccessibilityId("TextBox"));
+
+            var value = element.GetDomAttribute(attributeName);
+
+            Assert.That(value, Is.EqualTo(expectedValue));
+        }
+
+        [Test]
+        public void GetAttribute_Recognises_Pattern_Property()
```
Comment (aristotelos) [inline src/FlaUI.WebDriver.UITests/ElementTests.cs:259]: To comply with `MethodUnderTest_Condition_ExpectedOutcome`, rename to:
```suggestion
        public void GetAttribute_PatternProperty_ReturnsValue()
```

**Review by aristotelos — APPROVED** [review]
Approved with one minor test naming suggestion

**Comment by grokys** [comment]
Thanks, applied!
